### PR TITLE
Upgrade IPLD-prime to latest

### PR DESCRIPTION
**Pull request by hannahhoward**
# Goals

Upgrade to go-ipld-prime & graphsync to latest

# Implementation

- various rename to fit new node assembler syntax
- largely no other substantive changes

### CODE REVIEW

File: retrievalmarket/impl/blockio/traverser.go [hunk 1]
```diff
@@ -93,22 +93,28 @@ func (t *Traverser) Start(ctx context.Context) {
...
 			return
 		}
 		_ = traversal.Progress{
 			Cfg: &traversal.Config{
-				Ctx:                    ctx,
-				LinkLoader:             loader,
-				LinkNodeBuilderChooser: chooser,
+				Ctx:                        ctx,
+				LinkLoader:                 loader,
+				LinkTargetNodeStyleChooser: chooser,
```
Comment (shannonwells) [inline retrievalmarket/impl/blockio/traverser.go:117]: Nit: can this be a shortened at all?

File: shared_testutil/mocknet.go [hunk 2]
```diff
@@ -38,25 +38,27 @@ import (
...
-	Host1          host.Host
-	Host2          host.Host
-	AllSelector    ipld.Node
-	OrigBytes      []byte
+	Ctx              context.Context
+	Ds1              datastore.Batching
+	Ds2              datastore.Batching
+	DTStoredCounter1 *storedcounter.StoredCounter
+	DTStoredCounter2 *storedcounter.StoredCounter
+	StoredCounter1   *storedcounter.StoredCounter
```
Comment (shannonwells) [inline shared_testutil/mocknet.go:46]: Can this be called DIDStoredCounter1 & 2 like the ones above? or DealStoredCounter1 &2?

File: shared_testutil/test_ipld_tree.go [hunk 3]
```diff
@@ -68,31 +67,44 @@ func NewTestIPLDTree() TestIPLDTree {
...
 		leafBetaBlock, _                = blocks.NewBlockWithCid(storage[leafBetaLnk], leafBetaLnk.(cidlink.Link).Cid)
-		middleMapNode, middleMapNodeLnk = encode(fnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
-			mb.Insert(knb.CreateString("foo"), vnb.CreateBool(true))
-			mb.Insert(knb.CreateString("bar"), vnb.CreateBool(false))
-			mb.Insert(knb.CreateString("nested"), vnb.CreateMap(func(mb fluent.MapBuilder, knb fluent.NodeBuilder, vnb fluent.NodeBuilder) {
-				mb.Insert(knb.CreateString("alink"), vnb.CreateLink(leafAlphaLnk))
-				mb.Insert(knb.CreateString("nonlink"), vnb.CreateString("zoo"))
-			}))
+		middleMapNode, middleMapNodeLnk = encode(fluent.MustBuildMap(basicnode.Style.Map, 3, func(ma fluent.MapAssembler) {
+			nva := ma.AssembleEntry("foo")
```
Comment (shannonwells) [inline shared_testutil/test_ipld_tree.go:75]: You can actually chain these and cut your lines almost in half. I just made the changes and all the tests still pass:
```suggestion
			ma.AssembleEntry("foo").AssignBool(true)
			ma.AssembleEntry("bar").AssignBool(false)
			ma.AssembleEntry("nested").CreateMap(2, func(ma fluent.MapAssembler) {
				ma.AssembleEntry("alink").AssignLink(leafAlphaLnk)
				ma.AssembleEntry("nonlink").AssignString("zoo")
			})
		}))
		middleMapBlock, _                 = blocks.NewBlockWithCid(storage[middleMapNodeLnk], middleMapNodeLnk.(cidlink.Link).Cid)
		middleListNode, middleListNodeLnk = encode(fluent.MustBuildList(basicnode.Style.List, 4, func(la fluent.ListAssembler) {
			la.AssembleValue().AssignLink(leafAlphaLnk)
			la.AssembleValue().AssignLink(leafAlphaLnk)
			la.AssembleValue().AssignLink(leafBetaLnk)
			la.AssembleValue().AssignLink(leafAlphaLnk)
		}))
		middleListBlock, _    = blocks.NewBlockWithCid(storage[middleListNodeLnk], middleListNodeLnk.(cidlink.Link).Cid)
		rootNode, rootNodeLnk = encode(fluent.MustBuildMap(basicnode.Style.Map, 4, func(ma fluent.MapAssembler) {
			ma.AssembleEntry("plain").AssignString("olde string")
			ma.AssembleEntry("linkedString").AssignLink(leafAlphaLnk)
			ma.AssembleEntry("linkedMap").AssignLink(middleMapNodeLnk)
			ma.AssembleEntry("linkedList").AssignLink(middleListNodeLnk)
		}))
```

File: retrievalmarket/impl/blockio/traverser.go (hunk 1 above)
Comment (hannahhoward) [reply retrievalmarket/impl/blockio/traverser.go:117]: File under: requests for @warpfork

File: shared_testutil/test_ipld_tree.go (hunk 3 above)
Comment (hannahhoward) [reply shared_testutil/test_ipld_tree.go:75]: cool! cc: @warpfork this makes for a wonderfully concise syntax

File: shared_testutil/mocknet.go (hunk 2 above)
Comment (hannahhoward) [reply shared_testutil/mocknet.go]: done

File: shared_testutil/test_ipld_tree.go (hunk 3 above)
Comment (warpfork) [reply shared_testutil/test_ipld_tree.go]: oh thank goodness

File: retrievalmarket/impl/blockio/traverser.go (hunk 1 above)
Comment (warpfork) [reply retrievalmarket/impl/blockio/traverser.go:117]: I confess I hate this one too, but it does what it says on the tin.  I haven't thought of a shorter name.

**Pull request merged**